tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -294,27 +295,52 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential.FromComponents(
                 mechanism: "SCRAM-SHA-256", source: null, username: uniqueUnicodeUserName, password: unicodePassword);
-            
+
             AssertAuthenticationSucceeds(settings);
         }
 #endif
-        
+
+        [SkippableFact]
+        public void Authentication_succeeds_with_MONGODB_X509_mechanism()
+        {
+            RequireServer.Check().Tls(required: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH", isDefined: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD", isDefined: true);
+            var pathToClientCertificate = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH");
+            var password = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD");
+            var clientCertificate = new X509Certificate2(pathToClientCertificate, password);
+            var userName = GetRfc2253FormattedUsernameFromX509ClientCertificate(clientCertificate);
+            DropDatabaseUser(DriverTestConfiguration.Client, userName, databases: new[] { "$external" });
+            CreateX509DatabaseUser(DriverTestConfiguration.Client, userName);
```

Comment:
Turns out we need to manually add the user: mongo orchestration does add an X509 user, but I believe it's NOT the one in drivers-evergreen-tools: I believe Mongo orchestration instead creates a user with the client.pem in the mongo-orchestration library: (based off the default subject defined here: https://github.com/10gen/mongo-orchestration/blob/3c3e4890a1cfd7e8ba8ad795784b937ce3e23d54/mongo_orchestration/common.py#L38) which is NOT the subject of the `client.pem` in drivers-evergreen-tools (which is is `"C=US,ST=New York,L=New York City,O=MDB,OU=Drivers,CN=client`)